lib/ansible/galaxy/collection/__init__.py

```diff
@@ -956,6 +957,54 @@ def verify_collections(
     return results
 
 
+def uninstall_collections(requirements: list[Requirement], search_paths: list[str], skip_prompt: bool, artifacts_manager: ConcreteArtifactsManager = None):
+    """Uninstall collections in any of the search paths that match a requirement."""
+    collections = []
+    for requirement in requirements:
+        namespace, name = requirement.fqcn.split(".")
+        for installed in find_existing_collections(
+            search_paths,
+            artifacts_manager,
+            namespace_filter=namespace,
+            collection_filter=name,
```

Comment:
Why not access the properties directly?
```suggestion
        for installed in find_existing_collections(
            search_paths,
            artifacts_manager,
            namespace_filter=requirement.namespace,
            collection_filter=requirement.name,
```